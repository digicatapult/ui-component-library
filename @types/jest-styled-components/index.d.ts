declare module 'jest-styled-components/serializer'
