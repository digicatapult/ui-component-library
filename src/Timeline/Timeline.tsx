import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Button from './index.js'

describe('Input', () => {
  /*
  test('default', () => {
    const instance = renderer.create(<Button name="default">default</Button>)
    const tree = instance.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('style overrides', () => {
    const instance = renderer.create(
      <Button name="override-a" styles={{ height: '100px', background: 'red' }}>
        press me
      </Button>,
    )
    const tree = instance.toJSON()
    expect(tree).toMatchSnapshot()
  })
  */
})
