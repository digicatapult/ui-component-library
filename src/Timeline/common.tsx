import React from 'react'
import styled from 'styled-components'

const Ul = styled('ul')`
  list-style: none;
  margin: 0px;
  padding: 0px;
`

const Title = styled('h2')`
  display: flex;
  margin-top: 0px;
  flex-direction: row;
  justify-content: space-between;
  font-size: 16px;
  font-weight: normal;
  color: #bdc3c7;
  align-items: center;
  padding: 5px 5px;
  border-radius: 60px;
  color: ${(props: React.CSSProperties) => props.color || 'black'};
  border: ${(props: React.CSSProperties) => props.border || 'none'};
  opacity: 0.3;
`

const Status = styled('header')`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  gap: 10px;
  text-transform: uppercase;
  color: ${(props: React.CSSProperties) => props.color || 'black'};
  text-align: right;
  font-size: 12px;
`

const Label = styled('label')`
  min-height: 100px;
  width: 100%;
  display: block;
  color: ${(props: React.CSSProperties) => props.color || '#a7a7a7'};
  font-size: 11px;
  font-weight: 300;
  cursor: pointer;
  position: relative;
  padding: 0px 10px 0px 50px;
  box-sizing: border-box;
  overflow: hidden;

  &:hover {
    background: rgba(255, 125, 125, 0.2);
  }

  &::before {
    content: '';
    width: 12px;
    height: 12px;
    opacity: 0.3;
    display: block;
    position: absolute;
    left: 20px;
    top: 10px;
    background: ${(props: React.CSSProperties) => props.color || 'black'};
    transition: border 0.7s ease;
    border-radius: 100%;
    z-index: 99999;
  }

  &::after {
    content: '';
    width: 2px;
    opacity: 0.3;
    height: 100px;
    background: ${(props: React.CSSProperties) => props.color || 'black'};
    position: absolute;
    left: 25px;
    top: 0px;
  }
`

const Input = styled('input')`
  display: none;
`

const Container = styled('div')`
  overflow: hidden;
  left: 0px;
  padding: 5px;
  background: ${(props: React.CSSProperties) => props.background || 'none'};

  ${Input}:checked + ${Label} ${Title} {
    opacity: 1;
  }

  ${Input}:checked + ${Label} {
    &:hover {
      background: rgba(125, 255, 125, 0.2);
    }
    max-height: 300px;
  }

  ${Input}:checked + ${Label}:before {
    opacity: 1;
  }

  ${Input}:checked + ${Label}:after {
    opacity: 1;
  }
`
const Li = styled('li')`
  &:first-child ${Label}:before {
    top: 10px;
  }
  &:first-child ${Label}:after {
    top: 20px;
    height: 200px;
  }

  &:last-child ${Label}:before {
    top: 30%;
  }
  &:last-child ${Label}:after {
    top: -70px;
  }
`

const Heading = styled('h1')`
  padding: 0px 10px;
  font-family: Roboto;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  color: ${(props: React.CSSProperties) => props.color || 'black'};

  display: flex;
  flex-direction: row;
  gap: 10px;
  align-item: center;
  text-transform: uppercase;
`
const BlankText = styled('div')`
  width: 100%;
  height: 8px;
  background: rgba(0, 0, 0, 0.1);
`

const IconWrapper = styled('div')`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 25px;
  width: 25px;
  border: ${(props: React.CSSProperties) => props.border || 'none'};
  border-radius: 60px;
`

export {
  Heading,
  Li,
  Ul,
  Title,
  Label,
  Container,
  Input,
  BlankText,
  Status,
  IconWrapper,
}
