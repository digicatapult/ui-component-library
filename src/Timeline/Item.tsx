import React, { PropsWithChildren } from 'react'

import Checked from '../Icons/Checked.js'
import InProgress from '../Icons/InProgress.js'
import { Li, Input, Title, Label, Status, BlankText, IconWrapper } from './common.js'
import { variants } from './index.js'

export interface ItemProps {
  title: string
  checked: boolean
  variant?: any
  color?: string
  subStatus?: 'pending' | 'completed' | 'calculating' | ('submitted' & string)
  action?: () => void
}

const renderBlankText = () => <>
  <BlankText />
  <br />
  <BlankText />
</>

const Item: React.FC<PropsWithChildren<ItemProps>> = ({
  children,
  ...props
}: PropsWithChildren<ItemProps>) => {
    const styling: any = props?.variant ? variants[props.variant as 'hyproof'] : {}
  return (
    <Li>
      <Input type={'checkbox'} checked={props.checked} />
      <Label {...styling}>
        <Title {...styling}>
          {props.title}
          <Status {...styling}>
            {props.subStatus}
            <IconWrapper {...props}>
              {props.checked
                ? <Checked color={props.color} />
                : <InProgress color={props.color} />
              }
            </IconWrapper>
          </Status>
        </Title>
        {children ? (
          children
        ) : (
          renderBlankText()
        )}
      </Label>
    </Li>
  )
}

export default Item
