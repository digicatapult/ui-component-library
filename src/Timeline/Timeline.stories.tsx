import React from 'react'
import { action } from '@storybook/addon-actions'

import Timeline, { TimelineProps } from './index.js'

export default {
  title: 'Components/Timeline',
  component: Timeline,
  argTypes: {
    action: { action: 'click' },
    name: {
      control: 'text',
    },
  },
}

const items = [
  {
    title: 'A process has started',
    message: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    checked: true,
    action: action('click'), 
  },
  {
    title: 'Awaiting feedback',
    message: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    checked: false,
    action: action('click'), 
  },
  {
    title: 'Acknowledged',
    message: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    checked: false,
    action: action('click'), 
  },
]

const DefaultStoryTemplate = (args: TimelineProps) => (
  <Timeline {...args} name={'default'}>
    {items.map((itemProps) => <Timeline.Item {...itemProps} /> )}
  </Timeline>
)

export const Default = DefaultStoryTemplate.bind({})

Default.args = {
  type: 'submit',
}
