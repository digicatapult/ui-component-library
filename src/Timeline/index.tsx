import React from 'react'

import { Ul, Container, Heading } from './common.js'
import Item, { ItemProps } from './Item.js'

export interface TimelineProps {
  styles?: React.CSSProperties
  type?: 'submit'
  direction?: 'vertical' | 'horizontal'
  variant?: 'hyproof'
  name?: string
  onClick?: (e: any) => void
}

interface Timeline extends React.FC<React.PropsWithChildren<TimelineProps>> {
  Item: React.FC<React.PropsWithChildren<ItemProps>>
}

const Timeline: Timeline = Object.assign(
  (props: React.PropsWithChildren<TimelineProps>) => {
    const { children, ...rest } = props
    return (
      <Container {...props}>
        <Heading>{props.name}</Heading>
        <Ul>{props.children}</Ul>
      </Container>
    )
  },
  { Item },
)

export default Timeline
