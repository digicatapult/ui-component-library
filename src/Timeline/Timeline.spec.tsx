import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Timeline from './index.js'

describe.only('Timeline', () => {
  test('default', () => {
    const tree = renderer.create(
      <Timeline>
        <Timeline.Item
          title="test"
          checked={false}
        >
        </Timeline.Item>
      </Timeline>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
