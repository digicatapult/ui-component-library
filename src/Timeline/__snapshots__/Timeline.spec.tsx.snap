// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline default renders basic Timeline with a single Timeline.Item 1`] = `
.c2 {
  list-style: none;
  margin: 0px;
  padding: 0px;
}

.c9 {
  display: flex;
  margin-top: 0px;
  flex-direction: row;
  justify-content: space-between;
  font-size: 16px;
  font-weight: normal;
  color: #bdc3c7;
  align-items: center;
  padding: 5px 5px;
  border-radius: 60px;
  color: black;
  border: none;
  opacity: 0.3;
}

.c10 {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  gap: 10px;
  text-transform: uppercase;
  color: black;
  text-align: right;
  font-size: 12px;
}

.c7 {
  min-height: 100px;
  width: 100%;
  display: block;
  color: #a7a7a7;
  font-size: 11px;
  font-weight: 300;
  cursor: pointer;
  position: relative;
  padding: 0px 10px 0px 50px;
  box-sizing: border-box;
  overflow: hidden;
}

.c7:hover {
  background: rgba(255, 125, 125, 0.2);
}

.c7::before {
  content: '';
  width: 12px;
  height: 12px;
  opacity: 0.3;
  display: block;
  position: absolute;
  left: 20px;
  top: 10px;
  background: black;
  transition: border 0.7s ease;
  border-radius: 100%;
  z-index: 99999;
}

.c7::after {
  content: '';
  width: 2px;
  opacity: 0.3;
  height: 100px;
  background: black;
  position: absolute;
  left: 25px;
  top: 0px;
}

.c5 {
  display: none;
}

.c0 {
  overflow: hidden;
  left: 0px;
  padding: 5px;
  background: none;
}

.c0 .c4:checked+.c6 .c8 {
  opacity: 1;
}

.c0 .c4:checked+.c6 {
  max-height: 300px;
}

.c0 .c4:checked+.c6:hover {
  background: rgba(125, 255, 125, 0.2);
}

.c0 .c4:checked+.c6:before {
  opacity: 1;
}

.c0 .c4:checked+.c6:after {
  opacity: 1;
}

.c3:first-child .c6:before {
  top: 10px;
}

.c3:first-child .c6:after {
  top: 20px;
  height: 200px;
}

.c3:last-child .c6:before {
  top: 30%;
}

.c3:last-child .c6:after {
  top: -70px;
}

.c1 {
  padding: 0px 10px;
  font-family: Roboto;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  color: black;
  display: flex;
  flex-direction: row;
  gap: 10px;
  align-item: center;
  text-transform: uppercase;
}

.c12 {
  width: 100%;
  height: 8px;
  background: rgba(0, 0, 0, 0.1);
}

.c11 {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 25px;
  width: 25px;
  border: none;
  border-radius: 60px;
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <ul
    className="c2"
  >
    <li
      className="c3"
    >
      <input
        checked={false}
        className="c4 c5"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c6 c7"
      >
        <h2
          className="c8 c9"
        >
          timeline-item-basic
          <header
            className="c10"
          >
            <div
              className="c11"
            >
              <svg
                fill="none"
                height="18"
                viewBox="0 0 17 21"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 0C9.22674 2.05911e-08 10.9268 0.425849 12.4497 1.23983C13.9725 2.0538 15.2711 3.23078 16.2304 4.66651C17.1898 6.10224 17.7802 7.7524 17.9494 9.47082C18.1187 11.1892 17.8615 12.9229 17.2007 14.5182C16.5399 16.1135 15.4959 17.5212 14.1611 18.6166C12.8263 19.712 11.242 20.4614 9.54845 20.7982C7.85489 21.1351 6.1044 21.0491 4.45201 20.5479C2.79963 20.0466 1.29637 19.1456 0.075381 17.9246L7.5 10.5V0Z"
                  fill="black"
                />
              </svg>
            </div>
          </header>
        </h2>
        <div
          className="c12"
        />
        <br />
        <div
          className="c12"
        />
      </label>
    </li>
  </ul>
</div>
`;

exports[`Timeline default renders mixture of timelines complete/not-complete 1`] = `
.c2 {
  list-style: none;
  margin: 0px;
  padding: 0px;
}

.c9 {
  display: flex;
  margin-top: 0px;
  flex-direction: row;
  justify-content: space-between;
  font-size: 16px;
  font-weight: normal;
  color: #bdc3c7;
  align-items: center;
  padding: 5px 5px;
  border-radius: 60px;
  color: black;
  border: none;
  opacity: 0.3;
}

.c10 {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  gap: 10px;
  text-transform: uppercase;
  color: black;
  text-align: right;
  font-size: 12px;
}

.c7 {
  min-height: 100px;
  width: 100%;
  display: block;
  color: #a7a7a7;
  font-size: 11px;
  font-weight: 300;
  cursor: pointer;
  position: relative;
  padding: 0px 10px 0px 50px;
  box-sizing: border-box;
  overflow: hidden;
}

.c7:hover {
  background: rgba(255, 125, 125, 0.2);
}

.c7::before {
  content: '';
  width: 12px;
  height: 12px;
  opacity: 0.3;
  display: block;
  position: absolute;
  left: 20px;
  top: 10px;
  background: black;
  transition: border 0.7s ease;
  border-radius: 100%;
  z-index: 99999;
}

.c7::after {
  content: '';
  width: 2px;
  opacity: 0.3;
  height: 100px;
  background: black;
  position: absolute;
  left: 25px;
  top: 0px;
}

.c5 {
  display: none;
}

.c0 {
  overflow: hidden;
  left: 0px;
  padding: 5px;
  background: none;
}

.c0 .c4:checked+.c6 .c8 {
  opacity: 1;
}

.c0 .c4:checked+.c6 {
  max-height: 300px;
}

.c0 .c4:checked+.c6:hover {
  background: rgba(125, 255, 125, 0.2);
}

.c0 .c4:checked+.c6:before {
  opacity: 1;
}

.c0 .c4:checked+.c6:after {
  opacity: 1;
}

.c3:first-child .c6:before {
  top: 10px;
}

.c3:first-child .c6:after {
  top: 20px;
  height: 200px;
}

.c3:last-child .c6:before {
  top: 30%;
}

.c3:last-child .c6:after {
  top: -70px;
}

.c1 {
  padding: 0px 10px;
  font-family: Roboto;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  color: black;
  display: flex;
  flex-direction: row;
  gap: 10px;
  align-item: center;
  text-transform: uppercase;
}

.c12 {
  width: 100%;
  height: 8px;
  background: rgba(0, 0, 0, 0.1);
}

.c11 {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 25px;
  width: 25px;
  border: none;
  border-radius: 60px;
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <ul
    className="c2"
  >
    <li
      className="c3"
    >
      <input
        checked={false}
        className="c4 c5"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c6 c7"
      >
        <h2
          className="c8 c9"
        >
          timeline-item-not-complete
          <header
            className="c10"
          >
            <div
              className="c11"
            >
              <svg
                fill="none"
                height="18"
                viewBox="0 0 17 21"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 0C9.22674 2.05911e-08 10.9268 0.425849 12.4497 1.23983C13.9725 2.0538 15.2711 3.23078 16.2304 4.66651C17.1898 6.10224 17.7802 7.7524 17.9494 9.47082C18.1187 11.1892 17.8615 12.9229 17.2007 14.5182C16.5399 16.1135 15.4959 17.5212 14.1611 18.6166C12.8263 19.712 11.242 20.4614 9.54845 20.7982C7.85489 21.1351 6.1044 21.0491 4.45201 20.5479C2.79963 20.0466 1.29637 19.1456 0.075381 17.9246L7.5 10.5V0Z"
                  fill="black"
                />
              </svg>
            </div>
          </header>
        </h2>
        <div
          className="c12"
        />
        <br />
        <div
          className="c12"
        />
      </label>
    </li>
    <li
      className="c3"
    >
      <input
        checked={true}
        className="c4 c5"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c6 c7"
      >
        <h2
          className="c8 c9"
        >
          timeline-item-complete
          <header
            className="c10"
          >
            <div
              className="c11"
            >
              <svg
                fill="none"
                height="13"
                viewBox="0 0 17 13"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 5.5L6 10.5L15.5 1"
                  stroke="#a7a7a7"
                  strokeWidth="2.5"
                />
              </svg>
            </div>
          </header>
        </h2>
        <h1>
          header
        </h1>
      </label>
    </li>
  </ul>
</div>
`;

exports[`Timeline default renders text block if no children present in Timeline.Item 1`] = `
.c2 {
  list-style: none;
  margin: 0px;
  padding: 0px;
}

.c9 {
  display: flex;
  margin-top: 0px;
  flex-direction: row;
  justify-content: space-between;
  font-size: 16px;
  font-weight: normal;
  color: #bdc3c7;
  align-items: center;
  padding: 5px 5px;
  border-radius: 60px;
  color: black;
  border: none;
  opacity: 0.3;
}

.c10 {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  gap: 10px;
  text-transform: uppercase;
  color: black;
  text-align: right;
  font-size: 12px;
}

.c7 {
  min-height: 100px;
  width: 100%;
  display: block;
  color: #a7a7a7;
  font-size: 11px;
  font-weight: 300;
  cursor: pointer;
  position: relative;
  padding: 0px 10px 0px 50px;
  box-sizing: border-box;
  overflow: hidden;
}

.c7:hover {
  background: rgba(255, 125, 125, 0.2);
}

.c7::before {
  content: '';
  width: 12px;
  height: 12px;
  opacity: 0.3;
  display: block;
  position: absolute;
  left: 20px;
  top: 10px;
  background: black;
  transition: border 0.7s ease;
  border-radius: 100%;
  z-index: 99999;
}

.c7::after {
  content: '';
  width: 2px;
  opacity: 0.3;
  height: 100px;
  background: black;
  position: absolute;
  left: 25px;
  top: 0px;
}

.c5 {
  display: none;
}

.c0 {
  overflow: hidden;
  left: 0px;
  padding: 5px;
  background: none;
}

.c0 .c4:checked+.c6 .c8 {
  opacity: 1;
}

.c0 .c4:checked+.c6 {
  max-height: 300px;
}

.c0 .c4:checked+.c6:hover {
  background: rgba(125, 255, 125, 0.2);
}

.c0 .c4:checked+.c6:before {
  opacity: 1;
}

.c0 .c4:checked+.c6:after {
  opacity: 1;
}

.c3:first-child .c6:before {
  top: 10px;
}

.c3:first-child .c6:after {
  top: 20px;
  height: 200px;
}

.c3:last-child .c6:before {
  top: 30%;
}

.c3:last-child .c6:after {
  top: -70px;
}

.c1 {
  padding: 0px 10px;
  font-family: Roboto;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  color: black;
  display: flex;
  flex-direction: row;
  gap: 10px;
  align-item: center;
  text-transform: uppercase;
}

.c12 {
  width: 100%;
  height: 8px;
  background: rgba(0, 0, 0, 0.1);
}

.c11 {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 25px;
  width: 25px;
  border: none;
  border-radius: 60px;
}

<div
  className="c0"
>
  <h1
    className="c1"
  />
  <ul
    className="c2"
  >
    <li
      className="c3"
    >
      <input
        checked={false}
        className="c4 c5"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c6 c7"
      >
        <h2
          className="c8 c9"
        >
          timeline-item-3
          <header
            className="c10"
          >
            <div
              className="c11"
            >
              <svg
                fill="none"
                height="18"
                viewBox="0 0 17 21"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 0C9.22674 2.05911e-08 10.9268 0.425849 12.4497 1.23983C13.9725 2.0538 15.2711 3.23078 16.2304 4.66651C17.1898 6.10224 17.7802 7.7524 17.9494 9.47082C18.1187 11.1892 17.8615 12.9229 17.2007 14.5182C16.5399 16.1135 15.4959 17.5212 14.1611 18.6166C12.8263 19.712 11.242 20.4614 9.54845 20.7982C7.85489 21.1351 6.1044 21.0491 4.45201 20.5479C2.79963 20.0466 1.29637 19.1456 0.075381 17.9246L7.5 10.5V0Z"
                  fill="black"
                />
              </svg>
            </div>
          </header>
        </h2>
        <div
          className="c12"
        />
        <br />
        <div
          className="c12"
        />
      </label>
    </li>
  </ul>
</div>
`;

exports[`Timeline default renders timeline items with multiple children 1`] = `
.c2 {
  list-style: none;
  margin: 0px;
  padding: 0px;
}

.c9 {
  display: flex;
  margin-top: 0px;
  flex-direction: row;
  justify-content: space-between;
  font-size: 16px;
  font-weight: normal;
  color: #bdc3c7;
  align-items: center;
  padding: 5px 5px;
  border-radius: 60px;
  color: black;
  border: none;
  opacity: 0.3;
}

.c10 {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  gap: 10px;
  text-transform: uppercase;
  color: black;
  text-align: right;
  font-size: 12px;
}

.c7 {
  min-height: 100px;
  width: 100%;
  display: block;
  color: #a7a7a7;
  font-size: 11px;
  font-weight: 300;
  cursor: pointer;
  position: relative;
  padding: 0px 10px 0px 50px;
  box-sizing: border-box;
  overflow: hidden;
}

.c7:hover {
  background: rgba(255, 125, 125, 0.2);
}

.c7::before {
  content: '';
  width: 12px;
  height: 12px;
  opacity: 0.3;
  display: block;
  position: absolute;
  left: 20px;
  top: 10px;
  background: black;
  transition: border 0.7s ease;
  border-radius: 100%;
  z-index: 99999;
}

.c7::after {
  content: '';
  width: 2px;
  opacity: 0.3;
  height: 100px;
  background: black;
  position: absolute;
  left: 25px;
  top: 0px;
}

.c5 {
  display: none;
}

.c0 {
  overflow: hidden;
  left: 0px;
  padding: 5px;
  background: none;
}

.c0 .c4:checked+.c6 .c8 {
  opacity: 1;
}

.c0 .c4:checked+.c6 {
  max-height: 300px;
}

.c0 .c4:checked+.c6:hover {
  background: rgba(125, 255, 125, 0.2);
}

.c0 .c4:checked+.c6:before {
  opacity: 1;
}

.c0 .c4:checked+.c6:after {
  opacity: 1;
}

.c3:first-child .c6:before {
  top: 10px;
}

.c3:first-child .c6:after {
  top: 20px;
  height: 200px;
}

.c3:last-child .c6:before {
  top: 30%;
}

.c3:last-child .c6:after {
  top: -70px;
}

.c1 {
  padding: 0px 10px;
  font-family: Roboto;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  color: black;
  display: flex;
  flex-direction: row;
  gap: 10px;
  align-item: center;
  text-transform: uppercase;
}

.c11 {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 25px;
  width: 25px;
  border: none;
  border-radius: 60px;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    default
  </h1>
  <ul
    className="c2"
  >
    <li
      className="c3"
    >
      <input
        checked={true}
        className="c4 c5"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c6 c7"
      >
        <h2
          className="c8 c9"
        >
          timeline-item-1
          <header
            className="c10"
          >
            <div
              className="c11"
            >
              <svg
                fill="none"
                height="13"
                viewBox="0 0 17 13"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 5.5L6 10.5L15.5 1"
                  stroke="#a7a7a7"
                  strokeWidth="2.5"
                />
              </svg>
            </div>
          </header>
        </h2>
        <button>
          test me
        </button>
        <p>
          some paragraph
        </p>
        <h1>
          some header
        </h1>
        <div>
          some div
        </div>
        <span>
          some span
        </span>
      </label>
    </li>
    <li
      className="c3"
    >
      <input
        checked={true}
        className="c4 c5"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="c6 c7"
      >
        <h2
          className="c8 c9"
        >
          timeline-item-2
          <header
            className="c10"
          >
            <div
              className="c11"
            >
              <svg
                fill="none"
                height="13"
                viewBox="0 0 17 13"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 5.5L6 10.5L15.5 1"
                  stroke="#a7a7a7"
                  strokeWidth="2.5"
                />
              </svg>
            </div>
          </header>
        </h2>
        <h2>
          header
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </label>
    </li>
  </ul>
</div>
`;
